zulipterminal/model.py

```diff
@@ -559,6 +560,38 @@ def update_private_message(self, msg_id: int, content: str) -> bool:
         display_error_if_present(response, self.controller)
         return response["result"] == "success"
 
+    def toggle_topic_resolved_status(self, stream_id: int, topic_name: str) -> bool:
+        request: Dict[str, Any] = {
+            "anchor": "newest",
+            "num_before": 1,
+            "num_after": 0,
+            "narrow": [
+                {"operator": "sender", "operand": self.user_email},
```

Comment:
Agreed, I think removing this might make it possible to edit topic names even if you haven't sent any messages there.
Re, for topic edit permissions, I think we should do that before calling the toggle function. We should ideally not show them the button in the first place.

Reply:
True, if we can't resolve the topic, it would be confusing to bring it up. That said, in future we may want the topic info popup to be visible even if actions aren't possible, so we could take an approach a little like with the visual notifications checkbox in the stream popup when external notifications are not configured - that checkbox gets grayed out and we show red text to indicate why.